### ISSUE TITLE
 error: namespace "thrust" has no member "device_pointer_cast"

### ISSUE THREAD

**Issue by rickyHong** [NVIDIA/retinanet-examples#1]
~~> pip3 install .
Processing /host_volume/incheon/retinanet-examples
Requirement already satisfied: torch>=1.0.0a0 in /usr/local/lib/python3.5/dist-packages (from retinanet==0.1) (1.0.1.post2)
Requirement already satisfied: torchvision in /usr/local/lib/python3.5/dist-packages (from retinanet==0.1) (0.2.1)
Requirement already satisfied: apex@ git+https://github.com/NVIDIA/apex from git+https://github.com/NVIDIA/apex in /usr/local/lib/python3.5/dist-packages (from retinanet==0.1) (0.1)
Requirement already satisfied: pycocotools@ git+https://github.com/nvidia/cocoapi.git#subdirectory=PythonAPI from git+https://github.com/nvidia/cocoapi.git#subdirectory=PythonAPI in /root/.local/lib/python3.5/site-packages (from retinanet==0.1) (2.0)
Requirement already satisfied: pillow in /usr/local/lib/python3.5/dist-packages (from retinanet==0.1) (5.1.0)
Requirement already satisfied: requests in /usr/local/lib/python3.5/dist-packages (from retinanet==0.1) (2.20.1)
Requirement already satisfied: six in /usr/local/lib/python3.5/dist-packages (from torchvision->retinanet==0.1) (1.11.0)
Requirement already satisfied: numpy in /usr/local/lib/python3.5/dist-packages (from torchvision->retinanet==0.1) (1.14.5)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in /usr/local/lib/python3.5/dist-packages (from requests->retinanet==0.1) (1.22)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.5/dist-packages (from requests->retinanet==0.1) (2018.11.29)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/local/lib/python3.5/dist-packages (from requests->retinanet==0.1) (3.0.4)
Requirement already satisfied: idna<2.8,>=2.5 in /usr/local/lib/python3.5/dist-packages (from requests->retinanet==0.1) (2.7)
Building wheels for collected packages: retinanet
  Building wheel for retinanet (setup.py) ... error
  Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-5kq1wk60/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-lz3x_sfj --python-tag cp35:
  running bdist_wheel
  running build
  running build_py
  package init file 'retinanet/__init__.py' not found (or not a regular file)
  creating build
  creating build/lib.linux-x86_64-3.5
  creating build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/dali.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/infer.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/loss.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/model.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/data.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/utils.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/main.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/box.py -> build/lib.linux-x86_64-3.5/retinanet
  copying retinanet/train.py -> build/lib.linux-x86_64-3.5/retinanet
  creating build/lib.linux-x86_64-3.5/retinanet/backbones
  copying retinanet/backbones/layers.py -> build/lib.linux-x86_64-3.5/retinanet/backbones
  copying retinanet/backbones/utils.py -> build/lib.linux-x86_64-3.5/retinanet/backbones
  copying retinanet/backbones/__init__.py -> build/lib.linux-x86_64-3.5/retinanet/backbones
  copying retinanet/backbones/resnet.py -> build/lib.linux-x86_64-3.5/retinanet/backbones
  copying retinanet/backbones/fpn.py -> build/lib.linux-x86_64-3.5/retinanet/backbones
  running build_ext
  building 'retinanet._C' extension
  creating build/temp.linux-x86_64-3.5
  creating build/temp.linux-x86_64-3.5/csrc
  creating build/temp.linux-x86_64-3.5/csrc/cuda
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/lib/python3.5/dist-packages/torch/lib/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/torch/csrc/api/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/TH -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/THC -I/usr/local/cuda/include -I/usr/include/python3.5m -c csrc/extensions.cpp -o build/temp.linux-x86_64-3.5/csrc/extensions.o -std=c++11 -O2 -Wall -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/lib/python3.5/dist-packages/torch/lib/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/torch/csrc/api/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/TH -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/THC -I/usr/local/cuda/include -I/usr/include/python3.5m -c csrc/engine.cpp -o build/temp.linux-x86_64-3.5/csrc/engine.o -std=c++11 -O2 -Wall -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  /usr/local/cuda/bin/nvcc -I/usr/local/lib/python3.5/dist-packages/torch/lib/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/torch/csrc/api/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/TH -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/THC -I/usr/local/cuda/include -I/usr/include/python3.5m -c csrc/cuda/decode.cu -o build/temp.linux-x86_64-3.5/csrc/cuda/decode.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --compiler-options '-fPIC' -std=c++11 --expt-extended-lambda --use_fast_math -Xcompiler -Wall -arch=sm_60 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0
  /usr/local/cuda/bin/nvcc -I/usr/local/lib/python3.5/dist-packages/torch/lib/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/torch/csrc/api/include -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/TH -I/usr/local/lib/python3.5/dist-packages/torch/lib/include/THC -I/usr/local/cuda/include -I/usr/include/python3.5m -c csrc/cuda/nms.cu -o build/temp.linux-x86_64-3.5/csrc/cuda/nms.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --compiler-options '-fPIC' -std=c++11 --expt-extended-lambda --use_fast_math -Xcompiler -Wall -arch=sm_60 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0
  csrc/cuda/nms.cu(131): error: namespace "thrust" has no member "device_pointer_cast"

  1 error detected in the compilation of "/tmp/tmpxft_00007b69_00000000-6_nms.cpp1.ii".
  error: command '/usr/local/cuda/bin/nvcc' failed with exit status 1

What's wrong ?

**Comment by jmsinusa** [comment]
How did you get this error? Building the container? Do you have CUDA installed?

**Comment by rickyHong** [comment]
nvidia-docker run --rm --ipc=host -it nvcr.io/nvidia/pytorch:19.02-py3
